lib/api/apiUtils/bucket/bucketCors.js

```diff
@@ -0,0 +1,278 @@
+        if (!id) {
+            return undefined;
+        }
+        if (!Array.isArray(id) || id.length !== 1
+        || typeof id[0] !== 'string') {
+            return errors.MalformedXML;
+        }
+        if (id[0] === '') {
+            return undefined;
+        }
+        // NOTE: AWS does not make this check, but the spec specifies IDs should
+        // be up to 255 characters in length.
+        // http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTcors.html
+        if (id[0].length > 255) {
+            return errors.InvalidRequest.customizeDescription(errMsg);
+        }
+        return true; // to indicate ID exists
+    },
+    validateMaxAgeSeconds(seconds) {
+        if (!seconds) {
+            return undefined;
+        }
+        if (!Array.isArray(seconds) || seconds.length !== 1) {
+            return errors.MalformedXML;
+        }
+        if (seconds[0] === '') {
+            return undefined;
+        }
+        const parsedValue = parseInt(seconds[0], 10);
+        const errMsg = `MaxAgeSeconds "${seconds[0]}" is not a valid value.`;
+        if (isNaN(parsedValue) || parsedValue < 0) {
+            return errors.MalformedXML.customizeDescription(errMsg);
+        }
+        return parsedValue;
+    },
+    validateNumberRules(length) {
+        if (length > 100) {
+            return errors.InvalidRequest.customizeDescription(errs.numberRules);
+        }
+        return undefined;
+    },
+    validateOriginAndMethodExist(allowedMethods, allowedOrigins) {
+        if (allowedOrigins && allowedMethods &&
+        Array.isArray(allowedOrigins) &&
+        Array.isArray(allowedMethods) &&
+        allowedOrigins.length > 0 &&
+        allowedMethods.length > 0) {
+            return undefined;
+        }
+        return errors.MalformedXML
+            .customizeDescription(errs.originAndMethodExist);
+    },
+    validateMethods(methods) {
+        let invalidMethod;
+        function isValidMethod(method) {
+            if (method !== 'GET' && method !== 'PUT' && method !== 'HEAD' &&
+            method !== 'POST' && method !== 'DELETE') {
+                invalidMethod = method;
+                return false;
+            }
+            return true;
+        }
+        if (!methods.every(isValidMethod)) {
+            const errMsg = 'Found unsupported HTTP method in CORS config. ' +
+            `Unsupported method is "${invalidMethod}"`;
+            return errors.InvalidRequest.customizeDescription(errMsg);
+        }
+        return undefined;
+    },
+    validateOrigins(origins) {
+        for (let i = 0; i < origins.length; i++) {
+            const origin = origins[i];
+            const errMsg = `AllowedOrigin "${origin}" can not have ` +
+            'more than one wildcard.';
+            if (typeof origin !== 'string' || origin === '') {
+                return errors.MalformedXML;
+            }
+            const numberWildcards = (origin.match(/\*/g) || []).length;
```

Comment:
I made a utility function using indexOf but have yet to experience the fun of testing to see which is faster. (I assume the utility function but it will be exciting to witness).